multimodal/src/autogluon/multimodal/data/datamodule.py

```diff
@@ -31,6 +32,7 @@ def __init__(
         predict_data: Optional[pd.DataFrame] = None,
         id_mappings: Optional[Union[Dict[str, Dict], Dict[str, pd.Series]]] = None,
         val_use_training_mode: bool = False,
+        model_config: Optional[str] = None,
```

Comment:
`model_config` doesn't fit the context of datamodule. A better way may be trying to build the special dataset outside and pass it to argument `dataset`. We can detect whether `dataset` is None. Calling it in `set_dataset ` would be the same as calling `BaseDataset`.

Reply:
This new dataset is only used in training stage, and also this new dataset need model_config as input. An alternative way is to build a new data module.

Reply:
How about creating the special dataset in `_fit()` of predictor.py and passing it to datamodule initialization?

Reply:
Sure, it works for the current case, but it may not work if we have similar issues in the future.